libcudacxx/include/cuda/std/detail/libcxx/include/__utility/pair.h

```diff
@@ -69,707 +70,704 @@
-        second = _CUDA_VSTD::forward<second_type>(__p.second);
-        return *this;
-    }
-
-#if defined(__cuda_std__) && !defined(__CUDACC_RTC__)
-    _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_HOST constexpr
-    const pair& operator=(::std::pair<_T1, _T2>&& __p) const
-      noexcept(is_nothrow_assignable_v<const first_type&, first_type> &&
-               is_nothrow_assignable_v<const second_type&, second_type>)
-      requires(is_assignable_v<const first_type&, first_type> &&
-               is_assignable_v<const second_type&, second_type>) {
-        first = _CUDA_VSTD::forward<first_type>(__p.first);
-        second = _CUDA_VSTD::forward<second_type>(__p.second);
-        return *this;
-    }
-#endif // defined(__cuda_std__) && !defined(__CUDACC_RTC__)
-
-    template<class _U1, class _U2>
-    _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY constexpr
-    const pair& operator=(const pair<_U1, _U2>& __p) const
-      requires(is_assignable_v<const first_type&, const _U1&> &&
-               is_assignable_v<const second_type&, const _U2&>) {
-        first = __p.first;
-        second = __p.second;
-        return *this;
-    }
-
-#if defined(__cuda_std__) && !defined(__CUDACC_RTC__)
-    template<class _U1, class _U2>
-    _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_HOST constexpr
-    const pair& operator=(const ::std::pair<_U1, _U2>& __p) const
-      requires(is_assignable_v<const first_type&, const _U1&> &&
-               is_assignable_v<const second_type&, const _U2&>) {
-        first = __p.first;
-        second = __p.second;
-        return *this;
-    }
-#endif // defined(__cuda_std__) && !defined(__CUDACC_RTC__)
-
-    template<class _U1, class _U2>
-    _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY constexpr
-    const pair& operator=(pair<_U1, _U2>&& __p) const
-      requires(is_assignable_v<const first_type&, _U1> &&
-               is_assignable_v<const second_type&, _U2>) {
-        first = _CUDA_VSTD::forward<_U1>(__p.first);
-        second = _CUDA_VSTD::forward<_U2>(__p.second);
-        return *this;
-    }
-
-#if defined(__cuda_std__) && !defined(__CUDACC_RTC__)
-    template<class _U1, class _U2>
-    _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_HOST constexpr
-    const pair& operator=(::std::pair<_U1, _U2>&& __p) const
-      requires(is_assignable_v<const first_type&, _U1> &&
-               is_assignable_v<const second_type&, _U2>) {
-        first = _CUDA_VSTD::forward<_U1>(__p.first);
-        second = _CUDA_VSTD::forward<_U2>(__p.second);
-        return *this;
-    }
-#endif // defined(__cuda_std__) && !defined(__CUDACC_RTC__)
+  _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY constexpr const pair& operator=(pair const& __p) const
+    noexcept(_LIBCUDACXX_TRAIT(is_nothrow_copy_assignable, const _T1)
+             && _LIBCUDACXX_TRAIT(is_nothrow_copy_assignable, const _T2))
+    requires(is_copy_assignable_v<const first_type> && is_copy_assignable_v<const second_type>)
```

Comment:
In this hunk, the previous code uses `{first,second}_type` consistently, but you're now using both `{first,second}_type` and `_T{1,2}` in all (I think?) signatures that include a concept check, please settle on one.